db/commitlog/commitlog.hh

```diff
@@ -108,7 +108,7 @@ public:
 
         bool use_o_dsync = false;
         bool warn_about_segments_left_on_disk_after_shutdown = true;
-        bool allow_going_over_size_limit = true;
+        bool allow_going_over_size_limit = false;
```

Comment:
Which uses of the commitlog in the codebase will this line affect? Why do we need this line? Why changing `commitlog_use_hard_size_limit` to deprecated isn't sufficient?

Reply:
I think it affects only the tests commitlog_test and hint_test which create a commitlog without specifying `allow_going_over_size_limit` (some commitlog testcases do).
For other uses - the schema commitlog and hint manager set it explicitly to `true`, and the db commitlog gets the value from the parameter `commitlog_use_hard_size_limit`, which defaults to hard limit.
So it's not really required for deprecating the user parameter, but the purpose was that we want to move towards using the hard limit in general, and maybe removing the soft limit eventually, so if someone creates another commitlog it defaults to hard limit.